pkg/apiserver/rest/usecase/config.go

```diff
@@ -0,0 +1,191 @@
+/*
+Copyright 2022 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package usecase
+
+import (
+	"context"
+	"sync"
+
+	"github.com/oam-dev/kubevela/pkg/apiserver/model"
+
+	"github.com/pkg/errors"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/oam-dev/kubevela/apis/core.oam.dev/v1beta1"
+	"github.com/oam-dev/kubevela/apis/types"
+	"github.com/oam-dev/kubevela/pkg/apiserver/clients"
+	apis "github.com/oam-dev/kubevela/pkg/apiserver/rest/apis/v1"
+	"github.com/oam-dev/kubevela/pkg/definition"
+	"github.com/oam-dev/kubevela/pkg/utils/apply"
+)
+
+const (
+	labelVal              = "velacore-config"
+	definitionAlias       = definition.UserPrefix + "alias.config.oam.dev"
+	definitionType        = definition.UserPrefix + "type.config.oam.dev"
+	terraformProviderType = "terraform-provider"
+	// LabelProject is the label to store project name in an application
+	LabelProject = "project"
+)
+
+// ConfigHandler handle CRUD of configs
+type ConfigHandler interface {
+	ListConfigTypes(ctx context.Context, query string) ([]*apis.ConfigType, error)
+	GetConfigType(ctx context.Context, configType string) (*apis.ConfigType, error)
+	CreateConfig(ctx context.Context, req apis.CreateApplicationRequest) (*apis.ApplicationBase, error)
```

Comment:
CreateApplicationRequest?  Please redefine the request and response structs.

Reply:
It's indeed an application request and I will create an application based on the request.